packages/sku/config/typescript/tsconfig.js

```diff
@@ -24,14 +24,25 @@ module.exports = () => {
       // Fixes https://github.com/cypress-io/cypress/issues/1087
       skipLibCheck: true,
 
-      esModuleInterop: true,
+      // When dependencies have `type: module` TypeScript doesn't know how to resolve them,
+      // so we need to set this to `bundler` or `node`
+      moduleResolution: 'bundler',
+      // This is required by `bundler` and for dynamic `import()`
+      module: 'es2022',
+
+      // resolution-related
+      allowImportingTsExtensions: true,
```

Comment:
In [ESM mode](https://www.typescriptlang.org/docs/handbook/esm-node.html) extensions are mandatory. I find it weird importing from `.js` files when the file on disk is `.ts`.